clippy_utils/src/paths.rs

```diff
@@ -126,6 +126,27 @@ path_macros! {
     macro_path: PathNS::Macro,
 }
 
+// Paths in standard library (`alloc`/`core`/`std`, or against builtin scalar types)
+// should be added as diagnostic items directly into the standard library, through a
+// PR against the `rust-lang/rust` repository. If makes Clippy more robust in case
+// the item is moved around, e.g. if the library structure is reorganized.
+//
+// If their use is required before the next sync (which happens every two weeks),
+// they can be temporarily added below, prefixed with `DIAG_ITEM`, and commented
+// with a reference to the PR adding the diagnostic item:
+//
+// ```rust
+// // `sym::io_error_new` added in <https://github.com/rust-lang/rust/pull/142787>
+// pub static DIAG_ITEM_IO_ERROR_NEW: PathLookup = value_path!(std::io::Error::new);
+// ```
+//
+// During development, the "Added in …" comment is not required, but will make the
+// test fail once the PR is submitted and becomes mandatory to ensure that a proper
+// PR against `rust-lang/rust` has been created.
```

Comment:
One thing that's unclear to me is what happens if a Clippy PR that defines one such path is ready to be merged but the nightly hasn't been bumped yet so we don't have the diagnostic item.

Would that be considered a blocker for the Clippy PR and the PR author will need to wait for that bump to happen, and then remove the path here?

Or can the clippy PR be merged with the temporary path in this file? If this is the case, who migrates it to a diagnostic item? It seems like we would essentially be back to the current situation of people having to go through the paths every once in a while asynchronously anyway independent of the PR and the test wouldn't actually enforce the use of diagnostic items, would it?